test/topology_custom/test_mv_fail_updates.py

```diff
@@ -0,0 +1,68 @@
+#
+# Copyright (C) 2024-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+#
+import asyncio
+import pytest
+import time
+from test.topology.conftest import skip_mode
+from test.pylib.manager_client import ManagerClient
+
+from cassandra.cluster import ConsistencyLevel  # type: ignore
+from cassandra.query import SimpleStatement  # type: ignore
+
+
+async def wait_for_views(cql, node_count):
+    deadline = time.time() + 120
+    while time.time() < deadline:
+        done = await cql.run_async(f"SELECT COUNT(*) FROM system_distributed.view_build_status WHERE status = 'SUCCESS' ALLOW FILTERING")
+        if done[0][0] == node_count:
+            return
+        else:
+            time.sleep(0.2)
+    raise Exception("Timeout waiting for views to build")
+
+
+# This test makes sure that even if the view updates generation is failing on a node,
+# the reads can still succeed, as long as we can reach quorum.
+@pytest.mark.asyncio
+@skip_mode('release', "error injections aren't enabled in release mode")
+async def test_mv_fail_updates(manager: ManagerClient) -> None:
+    node_count = 3
+    servers = await manager.servers_add(node_count)
+    cql = manager.get_cql()
+    await cql.run_async(f"CREATE KEYSPACE ks WITH replication = {{'class': 'SimpleStrategy', 'replication_factor': 3}}")
+    await cql.run_async(f"CREATE TABLE ks.tab (key int, c int, PRIMARY KEY (key, c)) WITH SPECULATIVE_RETRY = 'ALWAYS'")
+    # Insert initial rows for building an index
+    for i in range(100):
+        await cql.run_async(f"INSERT INTO ks.tab (key, c) VALUES ({i}, 0)")
+
+    # The exception type used for the test is the one that is thrown when a replica write times out
+    # during the view update generation, but any exception other than the timeout is handled the same way.
+    await manager.api.enable_injection(servers[2].ip_addr, 'timeout_view_update_generation', one_shot=False)
+    await cql.run_async(f"CREATE INDEX tab_by_c ON ks.tab (c)")
+    # Insert more rows while building an index
```

Comment:
Is it important that you do these inserts "while" building the index? Because nothing actually guarantees that. These writes can happen after the index finished building, or before it started.